src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.NetNative.cs

```diff
@@ -6,19 +6,24 @@
 
 namespace System.Net.NetworkInformation
 {
-    public static class NetworkChange
+    public class NetworkChange
     {
+        //introduced for supporting design-time loading of System.Windows.dll
+        public static void RegisterNetworkChange(NetworkChange nc) { }
+
         public static event NetworkAddressChangedEventHandler NetworkAddressChanged;
 
+        public static event NetworkAvailabilityChangedEventHandler NetworkAvailabilityChanged;
+
         static NetworkChange()
         {
             RTConnectivity.NetworkInformation.NetworkStatusChanged += NetworkInformation_NetworkStatusChanged;
         }
 
-        private static void NetworkInformation_NetworkStatusChanged(Object sender)
+        private static void NetworkInformation_NetworkStatusChanged(object sender)
         {
             NetworkAddressChanged.Invoke(null, EventArgs.Empty);
-            return;
+            NetworkAvailabilityChanged.Invoke(null, new NetworkAvailabilityEventArgs(NetworkInterface.GetIsNetworkAvailable()));
```

Comment:
How are you testing the NetworkAddressChange.NetNative.cs changes you're making?  While "netcore50" target will build in CI, there are no "netcore50" target tests actually run in CI.

Are you running the "netcore50" tests manually?